lib/Extension/LanguageServerWorseReflection/DiagnosticProvider/WorseDiagnosticProvider.php

```diff
@@ -26,15 +29,22 @@ public function provideDiagnostics(TextDocumentItem $textDocument, CancellationT
         return call(function () use ($textDocument, $cancel) {
             $lspDiagnostics = [];
             foreach (yield $this->reflector->diagnostics(TextDocumentConverter::fromLspTextItem($textDocument)) as $diagnostic) {
-                $range = new Range(
-                    PositionConverter::byteOffsetToPosition($diagnostic->range()->start(), $textDocument->text),
-                    PositionConverter::byteOffsetToPosition($diagnostic->range()->end(), $textDocument->text),
-                );
+                /** @var Diagnostic $diagnostic */
+                $range = RangeConverter::toLspRange($diagnostic->range(), $textDocument->text);
+
                 $lspDiagnostic = ProtocolFactory::diagnostic($range, $diagnostic->message());
                 $lspDiagnostic->severity = self::toLspSeverity($diagnostic->severity());
                 $lspDiagnostic->source = 'phpactor';
-                $lspDiagnostics[] = $lspDiagnostic;
 
+                if ($diagnostic instanceof DeprecatedUsageDiagnostic) {
+                    $lspDiagnostic->tags[] = DiagnosticTag::DEPRECATED;
+                }
+
+                if ($diagnostic instanceof UnusedImportDiagnostic) {
```

Comment:
these "tags" should rather be added to the Diagnostic class, there can be many sources of "unnecessary" code